### ISSUE TITLE
Your Input class is unneeded.

### ISSUE THREAD

**Issue by gravypod** [BennyQBD/3DGameEngine#17]
I was watching your tutorial and noticed that you created the Input class witch is largely unneeded. You can do the same thing it does more efficiently with 

``` java
while (Keyboard.next()) {
    char key = Keyboard.getEventCharacter();
    boolean state = Keyboard.getEventKeyState(); // If the key is held down (true) or released (false)
   // TODO: Handle key clicks
}
```

Another thing to look into is using

``` java
int NUM_KEYCODES = Keyboard.getKeyCount();
```

instead of the current constant.

**Comment by agausmann** [comment]
Actually, in his case, the Input class is needed. Since there are multiple components potentially retrieving input, `Keyboard.next()` will not suffice - the first `Component.input()` called that uses a `while (Keyboard.next())` loop will hog all of the events, and none of the other components will be able to read them (unless they happen between the `input()` calls.
However, his Input class can be optimized using Keyboard.next() instead of checking every key indiviudally for its state.

**Comment by gravypod** [comment]
Sorry, I was more referring to the loop covering all of the keys. Iterating over every key is unneeded.

**Comment by agausmann** [comment]
I agree. If he had used `Keyboard.next()` instead it could potentially run faster.
